### PR TITLE
Fix changelog link

### PR DESCRIPTION
**Pull request by Souher**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

Changed changelog.md file such that tagged releases point to correct location
Fixes https://github.com/thanos-io/thanos/issues/5302

### CODE REVIEW

**Comment by Souher** [comment]
Hey @wiardvanrij, I've attempted to fix https://github.com/thanos-io/thanos/issues/5302. If you're happy with this change, i'll gladly also modify the rest of the tagged releases. 

Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -335,11 +337,11 @@ The binaries published with this release are built with Go1.17.8 to avoid [CVE-2
 
 ### Changed
 
--
+-
```
Comment (rajivharlalka) [inline CHANGELOG.md:340]: Can you try removing this white-space. CI must be failing for because of these white-space

**Review by rajivharlalka — COMMENTED** [review]
Have made some comments, can you try making these changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -103,7 +103,9 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 - [#5145](https://github.com/thanos-io/thanos/pull/5145) UI: Remove old Prometheus UI.
 
-## [v0.25.2](https://github.com/thanos-io/thanos/tree/release-0.25) - 2022.03.24
+## v0.25.2 - 2022.03.24
+
+Release [v0.25.2](https://github.com/thanos-io/thanos/releases/tag/v0.25.2)
```
Comment (yeya24) [inline CHANGELOG.md:108]: Please don't update the format. If the link is wrong just update the link here.

**Pull request closed without merging**